src/services/role/services/userPermissions.js

```diff
@@ -0,0 +1,38 @@
+const { userRoleHooks } = require('./userRoles');
+
+const filterPermissions = (userRoles, schoolPermissions) => {
+	const permissions = [];
+	userRoles.forEach((role) => {
+		if (Object.keys(schoolPermissions).includes(role)) {
+			Object.keys(schoolPermissions[role]).forEach((r) => {
```

Comment:
we usually use arrays to list the permissions of a role, why are you using object keys here?